nats-concepts/jetstream/source_and_mirror.md

```diff
@@ -0,0 +1,69 @@
+If you require the target stream to act as a read-only replica:
+* Configure the stream without listen subjects **or**
+* Temporarily disable the listen subjects through client authorizations. 
+{% endhint %}
+
+## Source specific
+A stream defining `Sources` is a generalized replication mechanism and allows for sourcing data from **one or more streams** concurrently as well as allowing direct write/publish by clients. Essentially the source streams and client writes are aggregated into a single interleaved stream.
+Subject transformation and filtering allow for powerful data distribution architectures.
+
+{% hint style="info" %}
+Sourcing messages does not retain sequence numbers. But it retain the in stream sequence of messages . Between streams sourced to the same target, the sequence of messages is undefined.
+{% endhint %}
+
+## Mirror specific
+A mirror can source its messages from **exactly one stream** and a clients can not directly write to the  mirror. Although messages cannot be published to a mirror directly by clients, messages can be deleted on-demand (beyond the retention policy), and consumers have all capabilities available on regular streams.
+
+{% hint style="info" %}
+* Mirrored messages retains the sequence numbers and timestamps of the origin stream. 
+* Mirrors can be used for for (geographic) load distribution with the `MirrorDirect` stream attribute. See: [https://docs.nats.io/nats-concepts/jetstream/streams#configuration](https://docs.nats.io/nats-concepts/jetstream/streams#configuration)
+
+{% endhint %}
+
+## Expected behavior in edge conditions
+
+* Source and mirror contracts are designed with one-way (geographic) data replication in mind. Neither configuration provides full synchronization between streams, which would include deletes or replication of other stream attributes.
+
+* The content of the stream from which a source or mirror is drawn needs to be reasonable stable. Quickly deleting messages after publishing them may result in inconsistent replication due to the asynchronous nature of the replication process.
+
+* Sources and Mirror try to be be efficient in replicating messages and are lenient towards the source/mirror origin temporarily not being reachable, e.g. when using leaf nodes, which are connected intermittently. For sake of efficiency the recovery interval in case of a disconnect is 10-20s.
+
+* Mirror and source agreements do not create a visible consumer in the origin stream.
+
+###  WorkQueue retention
+
+Source and mirror work with origin stream with workqueue retention. The source/mirror will act as a consumer removing messages from the origin stream. 
+
+The intended usage is for moving messages conveniently from one location to another (e.g. from a leaf node). It does not trivially implement a distributed workqueue. But a with the help of subject filtering can approximate a distributed workqueue.
+
```

Comment:
For your info (not sure how much we want to mention/advertise this feature very much at this point as it's working but not super-optimized in the current implementation) you can even have a working queue stream sourcing from another stream, and if that working queue has a limit _and_ a discard policy of 'new' it will work as a 'size bounded rolling buffer' type of thing - which is a pretty cool feature IMHO.

Reply:
We leave it out